internal/defact/default.go

```diff
@@ -0,0 +1,262 @@
+package defact
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	rt "runtime"
+	"strings"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/output"
+	"github.com/ActiveState/cli/internal/primer"
+	"github.com/ActiveState/cli/internal/strutils"
+	"github.com/ActiveState/cli/pkg/platform/runtime"
+	"github.com/gobuffalo/packr"
+	"github.com/thoas/go-funk"
+)
+
+type DefaultConfigurer interface {
+	Set(key string, value interface{})
+	GetString(key string) string
+}
+
+type exeFile struct {
+	fpath string
+	name  string
+	ext   string
+}
+
+func getDefaultProjectPath(d DefaultConfigurer) string {
+	return d.GetString("default_project_path")
+}
+
+func setDefaultProjectPath(d DefaultConfigurer, path string) {
+	d.Set("default_project_path", path)
+}
+
+type primable interface {
+	primer.Outputer
+}
+
+// isShimAndTargetsDir check if the specified filename is (probably) a shim and targets a file in the specified directory
+// This function is used during rollback to clean old shims
+func isShimAndTargetsDir(filename, dir string) bool {
+	contents, err := ioutil.ReadFile(filename)
+	if err != nil {
+		logging.Debug("Could not read file contents of shim candidate %s: %v", filename, err)
+		return false
+	}
+
+	targetRe := regexp.MustCompile("(?m)^(?:REM|#) State Tool Shim Target: (.*)$")
+	target := targetRe.FindStringSubmatch(string(contents))
+
+	if len(target) != 2 {
+		return false
+	}
+
+	if dir == "" {
+		return true
+	}
+
+	res, err := fileutils.PathContainsParent(target[1], dir)
```

Comment:
This doesn't make sense, we're just checking if the given path is in the cache dir? We should be checking if the path matched is the runtime directory that we are interested in. Whether it is in the cache dir is already determined by the fact that we know this is a shim and we know that the shim resides in our bin dir. We don't need an additional check to assert what we already know. We need to assert that this is a shim for our runtime dir.